Library/Homebrew/formula_installer.rb

```diff
@@ -707,31 +707,32 @@ def fetch_dependency(dep)
   sig { params(dep: Dependency, inherited_options: Options).void }
   def install_dependency(dep, inherited_options)
     df = dep.to_formula
-    tab = Tab.for_formula(df)
 
     if df.linked_keg.directory?
       linked_keg = Keg.new(df.linked_keg.resolved_path)
+      tab = Tab.for_keg(linked_keg)
       keg_had_linked_keg = true
       keg_was_linked = linked_keg.linked?
       linked_keg.unlink
     end
 
     if df.latest_version_installed?
       installed_keg = Keg.new(df.prefix)
+      tab ||= Tab.for_keg(linked_keg)
```

Comment:
I think this line is problematic for keg-only formulae, since `linked_keg` won't be defined for them, since they wouldn't have entered the `if df.linked_keg.directory?` block

Reply:
should this be `tab ||= Tab.for_keg(installed_keg)`?

Reply:
see #10761